### DISCUSSION TITLE
Timeline for full objects management.

### DISCUSSION THREAD

**Discussion by chiforbogdan** [keycloak/keycloak#26448] [Archive]
Hello,

As of today only KeycloakRealmImport CR is supported by the new operator in terms of Keycloak resources. Do you know what is the timeline/roadmap for supporting the multiple objects with full lifecycle (e.g. client, delete ream).

Thank you,

Bogdan Chifor.

**Comment by phinx110** [comment]
CRs for individual resources like realms, clients and users are of great importance to me because it greatly improves the ability to set up and update an environment. For this reason, I am still stuck on the legacy Keycloak operator.

There are two major reasons why I need this:
1. It automates the setup of realms, clients and users. Because the client and user credentials are exposed as secrets, they can be randomly generated and mounted by our apps. If not for the CRs, I would have to manually set up a new environment by clicking and coping from the GUI. This ends up being tedious and time-consuming at best and bug-prone at worst, especially when the project grows.
2. When developing a new app that requires Keycloack integrations, if it is done using CRs, the setup can be tested in a testing environment and then pushed to production. This creates reproducible updates, preventing problems where it works in test but not in production due to human error. There is no need to write additional update documentation when it is put into code using CRs.

Lots of big companies can throw excess employees at a problem, such as performing manual tasks. But, there are lots of smaller companies like mine that don't have many resources to spare. I am a one-man team keeping the project alive, but due to time pressure I am forced to use the legacy Keycloak operator because I really need CRs in order to work efficiently.

**Comment by vmuzikar** [comment]
Please see my answer here: https://github.com/keycloak/keycloak/discussions/27144#discussioncomment-8544819
